exp-models/addon/models/experiment.js

```diff
@@ -21,14 +26,32 @@ export default DS.Model.extend(JamModel, {
         return Ember.isEqual(this.get('active'), 'Active');
     }),
 
-    administrators: DS.hasMany('admin'),
-    history: DS.hasMany('history'),
-    sessions: DS.hasMany('session'),
-
     eligibilityCriteria: DS.attr('string'),
     eligibilityString: Ember.computed('eligibilityCriteria', function() {
         var eligibility = this.get('eligibilityCriteria');
         // TODO
         return eligibility || "None";
-    })
+    }),
+
+    history: DS.hasMany('history'),
+
+    sessionCollectionId: Ember.computed('shortId', function() {
+        // Return a string corresponding to the session collection shortID, to be used by model/adapter/serializer
+        // Eg an experiment called 'test0' would have a collection 'session-test0'
+        return `session${this.get('shortId')}s`; // FIXME: the spurious s is a requirement imposed by genschemas...
+    }),
+
+    _registerSessionModels() {
+        // Dynamically register the required models for a session table associated with this experiment
+        var cId = this.get('sessionCollectionId');
+        window.App.register(`model:${cId}`, SessionModel.extend()); // register a dummy model. This seems to work even if model already registered
```

Comment:
Is there a better workaround to this?  Using `window` in a built app makes me uneasy